### PR TITLE
Fix prefixing + formatting for enums

### PR DESCRIPTION
**Pull request by lu-zero**
A configuration such as
``` toml
[enum]
rename_variants = "ScreamingSnakeCase"
prefix_with_name = true
```

Should map an enum such `Foo::Bar` to `FOO_BAR` or `PREFIX_FOO_BAR` if there is a prefix set.

(I couldn't figure out how to move the former pr to this topic branch, sorry for the noise)

### CODE REVIEW

File: src/bindgen/rename.rs [hunk 1]
```diff
@@ -79,9 +79,22 @@ impl RenameRule {
             RenameRule::ScreamingSnakeCase => {
                 // Same as SnakeCase code above, but uses to_uppercase
                 let mut result = String::new();
+                let mut add_separator = true;
```
Comment (eqrion) [inline src/bindgen/rename.rs]: This is a bit confusing for me. Could you add a comment explaining what cases you're trying to handle. Additionally, do any other rename rules need to be updated to have compatible behaviors with consecutive upper case letters or underscores?

Comment (lu-zero) [reply src/bindgen/rename.rs]: Basically assume you have something with an `_` in the name for some reason, one being having collated a prefix as part of the `export.prefix = "Prefix"` combined with `enum.prefix_with_name = true`.

You have an enum such as
``` c
#[repr(C)]
pub enum Unprefixed {
    Some,
    Other,
}
```

You'd enjoy producing
``` c
typedef enum {
    PREFIX_UNPREFIXED_SOME,
    ...
}
```
and not
``` c
PREFIX_UNPREFIXED__SOME
```

The screaming-snake-case has also a corner case situation when you want to not add a `_` between two consecutive capital letters.

Comment (lu-zero) [reply src/bindgen/rename.rs]: SnakeCase would enjoy the same logic I guess.

Comment (eqrion) [reply src/bindgen/rename.rs]: Interesting. So I think this is okay. It's a bit hard to reason about these functions. In the long run, we should use a dedicated crate for this. I think [`heck`](https://github.com/withoutboats/heck) should support everything we need. The most important thing will probably be testing compatibility.

But again, this is probably good for now. Could you add a comment in the code explaining what this extra logic is doing?

Comment (lu-zero) [reply src/bindgen/rename.rs]: I reworked it a bit, is it fine like this?

**Pull request merged**